langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -192,12 +200,21 @@ async def acall_model(state: AgentState):
         # We return a list, because this will get added to the existing list
         return {"messages": [response]}
 
+    def add_retry_message(state: AgentState):
+        messages = state["messages"]
+        last_message = messages[-1]
+        retry_message = handle_parsing_errors(last_message)
+        if not isinstance(retry_message, list):
+            retry_message = [retry_message]
+        return {"messages": retry_message}
+
     # Define a new graph
     workflow = StateGraph(AgentState)
 
     # Define the two nodes we will cycle between
     workflow.add_node("agent", RunnableLambda(call_model, acall_model))
     workflow.add_node("action", ToolNode(tools))
+    workflow.add_node("retry", RunnableLambda(add_retry_message))
```

Comment:
alternative to adding a new node is to just process messages containing parsing issues inside the `call_model` functions. for instance, we could add an AIMessage functioning as a thought/observation, similar to AgentExecutor. but then in `should_continue` we would need to distinguish this case.